src/number.rs

```diff
@@ -219,6 +231,23 @@ impl Number {
             None
         }
     }
+    
+    /// Gets a hint on how the number is stored internally.
```

Comment:
I don't think we should conceptualize this as a hint on how the number is stored internally. The internal storage is not the user's business. Could you frame this in a way that does not involve internal storage?